src/language_features/code_action.rs

```diff
@@ -72,7 +72,11 @@ fn code_actions_for_ranges(
     {
         buff_diags
             .iter()
-            .filter(|(server_name, d)| ranges_overlap(d.range, ranges[server_name]))
+            .filter(|(server_name, d)| {
+                ranges
+                    .get(server_name)
+                    .is_some_and(|r| ranges_overlap(d.range, *r))
```

Comment:
I don't understand how this crash can happen.
`ranges` should always contain all eligible servers.
How to reproduce the problem?